boards/nxp/frdm_mcxn947/frdm_mcxn947_mcxn947_cpu0_qspi.dts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2024 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/dts-v1/;
+
+#include "frdm_mcxn947_mcxn947_cpu0.dtsi"
+
+/delete-node/ &slot0_partition;
+/delete-node/ &slot1_partition;
+/delete-node/ &storage_partition;
+
+/ {
+	model = "NXP FRDM_N94 board, QSPI variant";
+	compatible = "nxp,mcxn947", "nxp,mcx";
+
+	chosen {
+		zephyr,flash = &w25q64jvssiq;
+	};
+
+};
+
+&w25q64jvssiq {
+	partitions {
+		compatible = "fixed-partitions";
```

Comment:
Hi @mmahadevan108 
The https://github.com/mcu-tools/mcuboot/pull/2050/files will change MCUBoot FW update mode to Upgrade Only. It is only was needed for On-Chip flash.
But because the slot partitions are defined in external Flash with the write-block-size = 1, it is not needed. 
Could you comment why [it](https://github.com/mcu-tools/mcuboot/pull/2050/files is needed?

Reply:
Build failures were seen when building MCUBoot

Reply:
> Build failures were seen when building MCUBoot with the QSPI variant

This could be a potential bug.

Reply:
The MCUboot for the QSPI variant still builds with the internal flash as the chosen node and without this change we would see the build failures which is expected. 
Please take a look at this overlay file in MCUboot: https://github.com/mcu-tools/mcuboot/blob/main/boot/zephyr/boards/frdm_mcxn947_mcxn947_cpu0_qspi.overlay

Reply:
I agree that this might be a bug in mcuboot as its probably using chosen node to decide flash properties and not accounting for the slots to reside on a different flash. I can take a look at this.